include/SFML/System/Clock.hpp

```diff
@@ -107,7 +108,47 @@ class SFML_SYSTEM_API Clock
     /// \return Time elapsed
     ///
     ////////////////////////////////////////////////////////////
-    Time getElapsedTime() const;
+    [[nodiscard]] Time getElapsedTime() const noexcept;
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Check whether certain time has passed
+    ///
+    /// This function returns boolean value based on whether certain time has passed.
+    ///
+    /// \return True whether certain time has passed, false otherwise
+    ///
+    ////////////////////////////////////////////////////////////
+    [[nodiscard]] bool hasTimePassed(const Time& time) const noexcept;
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Start the clock
+    ///
+    /// This function starts the clock.
+    ///
+    /// \see pause, isRunning
+    ///
+    ////////////////////////////////////////////////////////////
+    void start() noexcept;
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Pause the clock
+    ///
+    /// This function pauses the clock.
+    ///
+    /// \see start, isPaused
+    ///
+    ////////////////////////////////////////////////////////////
+    void pause() noexcept;
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Toggle clock state
+    ///
+    /// This function toggles clock state between running and paused.
+    ///
+    /// \see start, pause, isRunning, isPaused
+    ///
+    ////////////////////////////////////////////////////////////
+    void toggle() noexcept;
```

Comment:
I read posts about `[[nodiscard]]`/`noexcept` and I understand that this PR may push this concept too much when its main goal is ability to pause a clock. Nonetheless, sooner or later these changes **may** be necessary but now for I'll reiterate changes for more consistency.

As for the point with size increase there was idea to seperate this feature and make `sf::PausableClock` along with unchanged `sf::Clock`. Then, user wouldn't have to pay for pause feature.

`toggle()` was motivated with simplifying code like below but since `setX(bool)` is more used in library and is more versatile I understand your point. It can substituted with `clock.setRunning(!clock.isRunning())` though.
```cpp
if (clock.isRunning())
{
    clock.pause();
}
else
{
    clock.start();
}
```

With addition of `setRunning(bool)` I'd drop `isPaused()` and keep `isRunning()` for consistency but keep `start()` and `pause()`.

I'll wait for responses and more feedback then commit accordingly ;)